packages/core/src/app/shipping/ConsignmentAddressSelector.tsx

```diff
@@ -73,11 +84,22 @@ const ConsignmentAddressSelector = ({
         }
 
         try {
-            await assignItem({
+            const {
+                data: { getConsignments },
+            } = await assignItem({
                 address,
                 lineItems: consignment.lineItems.map(({ id, quantity }) => ({ itemId: id, quantity })),
             });
 
+            const currentConsignments = getConsignments();
+
+            if (currentConsignments) {
```

Comment:
Do we need to check the length of consignments? Since `[]` is truthy as well?

Reply:
The condition is to make Typescript happy as `getConsignments` can return `undefined` by definition, not strictly required.
Maybe when creating a consignment fails, checking length here can skip `setRecommendedOrMissingShippingOption` and save some time. I have added the length check in this case.